packages/abi-gen/test-cli/fixtures/contracts/AbiGenDummy.sol

```diff
@@ -138,4 +138,134 @@ contract AbiGenDummy
     uint someState;
     function nonPureMethod() public returns(uint) { return someState += 1; }
     function nonPureMethodThatReturnsNothing() public { someState += 1; }
+
+    /// @dev complex input is dynamic and more difficult to decode than simple input.
+    struct ComplexInput {
+        uint256 foo;
+        bytes bar;
+        string car;
+    }
+
+    /// @dev complex input is dynamic and more difficult to decode than simple input.
+    struct ComplexOutput {
+        ComplexInput input;
+        bytes lorem;
+        bytes ipsum;
+        string dolor;
+    }
+
+    /// @dev The fallback function calls into this contract and executes one of the above functions.
+    /// This allows us to test `getABIDecodedTransactionData` and `getABIDecodedReturnData` that is
+    /// include in contract wrappers.
+    // solhint-disable no-complex-fallback
+    function ()
```

Comment:
Doesn't this function just loop forever? If the calldata hits the fallback function, then the fallback will call back into this contract using the same calldata, which would intuitively also make it hit the fallback function, creating an infinite loop. Please let me know if I'm wrong about this, but I've done a bit of testing locally and it seems to loop forever.

Reply:
Wouldn't this never get called if used as intended because the functions it's forwarding to are defined in this contract?

Reply:
Ah, yeah you're right this can be removed now that all tests are in the same contract. 🙏

Reply:
@dorothy-zbornak Yeah, I think that is the case. I was just saying that it would loop infinitely if it was called for some reason.